spec.md

```diff
@@ -0,0 +1,954 @@
+
+### Supervised Lifecycle Management
+
+* For Plugins packaged in software form:
+  * Plugin Packages SHOULD use a well-documented container image format (e.g., Docker, OCI).
+  * The chosen package image format MAY expose configurable Plugin properties as environment variables, unless otherwise indicated in the section below.
+    Variables so exposed SHOULD be assigned default values in the image manifest.
+  * A Plugin Supervisor MAY programmatically evaluate or otherwise scan a Plugin Package’s image manifest in order to discover configurable environment variables.
+  * A Plugin SHALL NOT assume that an operator or Plugin Supervisor will scan an image manifest for environment variables.
+
+#### Environment Variables
+
+* Variables defined by this specification SHALL be identifiable by their `CMI_` name prefix.
+* Configuration properties not defined by the CMI specification SHALL NOT use the same `CMI_` name prefix; this prefix is reserved for common configuration properties defined by the CMI specification.
+* The Plugin Supervisor SHOULD supply all RECOMMENDED CMI environment variables to a Plugin.
+* The Plugin Supervisor SHALL supply all REQUIRED CMI environment variables to a Plugin.
+
+##### `CMI_ENDPOINT`
+
+Network endpoint at which a Plugin SHALL host CMI RPC services. The general format is:
+
+    {scheme}://{authority}{endpoint}
+
+The following address types SHALL be supported by Plugins:
+
+    unix:///path/to/unix/socket.sock
+
+Note: All UNIX endpoints SHALL end with `.sock`. See [gRPC Name Resolution](https://github.com/grpc/grpc/blob/master/doc/naming.md).
+
+This variable is REQUIRED.
+
+#### Operational Recommendations
+
+The Plugin Supervisor expects that a Plugin SHALL act as a long-running service vs. an on-demand, CLI-driven process.
+
+Supervised plugins MAY be isolated and/or resource-bounded.
+
+##### Logging
+
+* Plugins SHOULD generate log messages to ONLY standard output and/or standard error.
+  * In this case the Plugin Supervisor SHALL assume responsibility for all log lifecycle management.
+* Plugin implementations that deviate from the above recommendation SHALL clearly and unambiguously document the following:
+  * Logging configuration flags and/or variables, including working sample configurations.
+  * Default log destination(s) (where do the logs go if no configuration is specified?)
+  * Log lifecycle management ownership and related guidance (size limits, rate limits, rolling, archiving, expunging, etc.) applicable to the logging mechanism embedded within the Plugin.
+* Plugins SHOULD NOT write potentially sensitive data to logs (e.g. secrets).
+
+##### Available Services
+
+* Plugin Packages MAY support all or a subset of CMI services; service combinations MAY be configurable at runtime by the Plugin Supervisor.
```

Comment:
Can we describe what happens if some of the services or RPCs are not supported?

Reply:
Okay.

Reply:
Not counting the identity service, we have only one service (machine) as of now. This may not be applicable to CMI. We can reword this as `Available RPCs`, but this is covered in detail in the preceding text anyway.

Reply:
Added it here - https://github.com/gardener/machine-spec/pull/11/files#diff-958e7270f96f5407d7d980f500805b1bR122